### ISSUE TITLE
flow context gets lost on deploy for nested subflow

### ISSUE THREAD

**Issue by jpirko** [node-red/node-red#3996]
### Current Behavior

When subflow is instantiated within another subflow, the flow.get/set does not work as expected. After deploy, flow.get() returns "undefined" for previously stored value. When subflow is instantiated directly, the behaviour is correct.

### Expected Behavior

I expect the flow.get/set to work correctly after deploy no matter where is the subflow instantiated.

### Steps To Reproduce

Create subflow within another subflow. The inner subflow should use flow.get/set to store and load values. After deploy, undefined is loaded instead of previously stored value.

The flow below has 2 instances of the same "load and store" subflow. One is instantiated directly, the second one using a "nest" subflow. I would expect the consistent outputs from both of these instances. As you can see in the debug console, the output differs after deploy.

### Example flow
```
[
    {
        "id": "c83b9d686d97aa70",
        "type": "tab",
        "label": "Flow 1",
        "disabled": false,
        "info": "",
        "env": []
    },
    {
        "id": "f5d0bc75059d2df2",
        "type": "subflow",
        "name": "load and store",
        "info": "",
        "category": "",
        "in": [
            {
                "x": 60,
                "y": 140,
                "wires": [
                    {
                        "id": "15f979b5de0720cc"
                    }
                ]
            }
        ],
        "out": [
            {
                "x": 680,
                "y": 140,
                "wires": [
                    {
                        "id": "15f979b5de0720cc",
                        "port": 0
                    }
                ]
            }
        ],
        "env": [],
        "meta": {},
        "color": "#DDAA99"
    },
    {
        "id": "c27441dc423ae67c",
        "type": "subflow",
        "name": "nest",
        "info": "",
        "category": "",
        "in": [
            {
                "x": 50,
                "y": 30,
                "wires": [
                    {
                        "id": "70c35626d1c73154"
                    }
                ]
            }
        ],
        "out": [
            {
                "x": 460,
                "y": 20,
                "wires": [
                    {
                        "id": "70c35626d1c73154",
                        "port": 0
                    }
                ]
            }
        ],
        "env": [],
        "meta": {},
        "color": "#DDAA99"
    },
    {
        "id": "15f979b5de0720cc",
        "type": "function",
        "z": "f5d0bc75059d2df2",
        "name": "function 1",
        "func": "const old = flow.get(\"value\");\nflow.set(\"value\", msg.payload);\nmsg.payload = old\nreturn msg;",
        "outputs": 1,
        "noerr": 0,
        "initialize": "",
        "finalize": "",
        "libs": [],
        "x": 300,
        "y": 140,
        "wires": [
            []
        ]
    },
    {
        "id": "70c35626d1c73154",
        "type": "subflow:f5d0bc75059d2df2",
        "z": "c27441dc423ae67c",
        "name": "",
        "x": 270,
        "y": 60,
        "wires": [
            []
        ]
    },
    {
        "id": "1958a2c7e0f1b70d",
        "type": "subflow:f5d0bc75059d2df2",
        "z": "c83b9d686d97aa70",
        "name": "",
        "x": 390,
        "y": 140,
        "wires": [
            [
                "b7598783a7a424e8"
            ]
        ]
    },
    {
        "id": "b7598783a7a424e8",
        "type": "debug",
        "z": "c83b9d686d97aa70",
        "name": "direct load and store",
        "active": true,
        "tosidebar": true,
        "console": false,
        "tostatus": false,
        "complete": "payload",
        "targetType": "msg",
        "statusVal": "",
        "statusType": "auto",
        "x": 620,
        "y": 140,
        "wires": []
    },
    {
        "id": "589f4692325717bb",
        "type": "inject",
        "z": "c83b9d686d97aa70",
        "name": "",
        "props": [
            {
                "p": "payload"
            },
            {
                "p": "topic",
                "vt": "str"
            }
        ],
        "repeat": "3",
        "crontab": "",
        "once": true,
        "onceDelay": 0.1,
        "topic": "",
        "payload": "",
        "payloadType": "date",
        "x": 130,
        "y": 160,
        "wires": [
            [
                "1958a2c7e0f1b70d",
                "e03ddec12675dfe5"
            ]
        ]
    },
    {
        "id": "e03ddec12675dfe5",
        "type": "subflow:c27441dc423ae67c",
        "z": "c83b9d686d97aa70",
        "name": "",
        "x": 370,
        "y": 200,
        "wires": [
            [
                "69921e8692b3b508"
            ]
        ]
    },
    {
        "id": "69921e8692b3b508",
        "type": "debug",
        "z": "c83b9d686d97aa70",
        "name": "nested load and store",
        "active": true,
        "tosidebar": true,
        "console": false,
        "tostatus": false,
        "complete": "payload",
        "targetType": "msg",
        "statusVal": "",
        "statusType": "auto",
        "x": 620,
        "y": 200,
        "wires": []
    }
]
```

### Environment

- Node-RED version: 3.0.2
- Node.js version: v19.2.0
- npm version: 9.2.0
- Platform/OS: MacOS 12.6
- Browser: Safari 16.0 (17614.1.25.9.10, 17614)

**Comment by jpirko** [comment]
Note that with context.get/set() the behaviour is the same, broken as well.

**Comment by knolleary** [comment]
Fixed via #4093 - will be in 3.1
